app/assets/javascripts/discourse/app/components/user-menu/profile-tab-content.js

```diff
@@ -26,8 +26,11 @@ export default class UserMenuProfileTabContent extends Component {
   get showToggleAnonymousButton() {
     return (
       (this.siteSettings.allow_anonymous_posting &&
-        this.currentUser.trust_level >=
-          this.siteSettings.anonymous_posting_min_trust_level) ||
+        this.currentUser.isInAnyGroups(
+          this.siteSettings.groupSettingArray(
+            "anonymous_posting_allowed_groups"
+          )
+        )) ||
```

Comment:
The APIs here feels a little too verbose from my perspective. I wonder if we can maybe simplify it to `this.siteSettings.userAllowed?("anonymous_posting_allowed_groups", this.currentUser)`

Reply:
Oh that is nice! I will make this change. IMO we could change it to `userInAnyGroups` to match this one https://github.com/discourse/discourse/blob/63a0700d45f755d0f432a9075ae7afbed9cd6ab0/app/assets/javascripts/discourse/app/models/user.js#L705-L717  . Also sadly no `?` for functions in JS land :cry: